Youtube/actions/updateChannel.ts

```diff
@@ -0,0 +1,291 @@
+import type { AppContext } from "../mod.ts";
+import getAccessToken from "../utils/getAccessToken.ts";
+
+export interface UpdateChannelOptions {
+  /**
+   * @description ID do canal a ser atualizado
+   */
+  channelId: string;
+
+  /**
+   * @description Título do canal (opcional)
+   */
+  title?: string;
+
+  /**
+   * @description Descrição do canal (opcional)
+   */
+  description?: string;
+
+  /**
+   * @description ID do vídeo para definir como trailer do canal (opcional)
+   */
+  unsubscribedTrailer?: string;
+
+  /**
+   * @description Tags do canal, separadas por vírgula ou array (opcional)
+   */
+  keywords?: string | string[];
+
+  /**
+   * @description País de origem do canal (código de duas letras, ex: BR) (opcional)
+   */
+  country?: string;
+
+  /**
+   * @description Idioma padrão do canal (código, ex: pt-BR) (opcional)
+   */
+  defaultLanguage?: string;
+
+  /**
+   * @description Tópicos relacionados ao canal (IDs de tópicos do Freebase/Wiki) (opcional)
+   */
+  topicCategories?: string[];
+
+  /**
+   * @description Token de acesso do YouTube (opcional)
+   */
+  tokenYoutube?: string;
+}
+
+interface UpdateChannelResult {
+  success: boolean;
+  message: string;
+  channel?: any;
+}
+
+/**
+ * @title Configurar Canal do YouTube
+ * @description Atualiza diversas informações do canal como descrição, trailer e outras configurações
+ */
+export default async function action(
+  props: UpdateChannelOptions,
+  req: Request,
+  _ctx: AppContext,
+): Promise<UpdateChannelResult> {
+  const {
+    channelId,
+    title,
+    description,
+    unsubscribedTrailer,
+    keywords,
+    country,
+    defaultLanguage,
+    topicCategories,
+    tokenYoutube,
+  } = props;
+
+  if (!channelId) {
+    console.error("ID do canal não fornecido");
+    return { success: false, message: "ID do canal é obrigatório" };
+  }
+
+  // Obter o token de acesso
+  const accessToken = tokenYoutube || getAccessToken(req);
+
+  if (!accessToken) {
+    console.error("Token de acesso não encontrado");
+    return { success: false, message: "Autenticação necessária" };
+  }
+
+  try {
```

Comment:
try catchs are too big on your functions. create a separate call for that logic, and wrap it inside here.

```
try {
   trueActionFn(props);
} catch (err) {
   // do something
}
```